src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -39,16 +40,17 @@ object EntityClient {
 
   case class ImportBagit(workspaceNamespace: String, workspaceName: String, bagitRq: BagitImportRequest)
 
-  def props(requestContext: RequestContext)(implicit executionContext: ExecutionContext): Props = Props(new EntityClient(requestContext))
+  def props(requestContext: RequestContext, modelSchema: ModelSchema)(implicit executionContext: ExecutionContext): Props = Props(new EntityClient(requestContext, modelSchema))
 
   def colNamesToAttributeNames(headers: Seq[String], requiredAttributes: Map[String, String]): Seq[(String, Option[String])] = {
     headers.tail map { colName => (colName, requiredAttributes.get(colName))}
   }
 
-  def backwardsCompatStripIdSuffixes(tsvLoadFile: TSVLoadFile, entityType: String): TSVLoadFile = {
-    ModelSchema.getTypeSchema(entityType) match {
-      case Failure(regrets) => tsvLoadFile
-      case Success(metaData) =>
+  def backwardsCompatStripIdSuffixes(tsvLoadFile: TSVLoadFile, entityType: String, modelSchema: ModelSchema): TSVLoadFile = {
+    modelSchema.isInstanceOf[FlexibleModelSchema] match {
+      case true => tsvLoadFile
+      case false =>
+        val metaData = modelSchema.getTypeSchema(entityType)
```

Comment:
To avoid possibly throwing an exception in this case (the FC schema case), you can put this is a `Try`:
```
        Try(modelSchema.getTypeSchema(entityType)) match {
          case Failure(_) => tsvLoadFile
          case Success(metaData) =>
            val newHeaders = tsvLoadFile.headers.map { header =>
              val headerSansId = header.stripSuffix("_id")
              if (metaData.requiredAttributes.keySet.contains(headerSansId) || metaData.memberType.contains(headerSansId)) {
                headerSansId
              } else {
                header
              }
            }
            tsvLoadFile.copy(headers = newHeaders)
        }
```
This should maintain the behavior of the function for this case.